pkg/controllers/multiclusteringress/mci_controller.go

```diff
@@ -91,9 +91,11 @@ func (c *MCIController) handleMCIDelete(ctx context.Context, mci *networkingv1al
 func (c *MCIController) handleMCICreateOrUpdate(ctx context.Context, mci *networkingv1alpha1.MultiClusterIngress) (controllerruntime.Result, error) {
 	klog.V(4).InfoS("Begin to handle multiClusterIngress create or update event", "namespace", mci.Namespace, "name", mci.Name)
 
-	finalizersUpdated := controllerutil.AddFinalizer(mci, MCIControllerFinalizer)
+	objPatch := client.MergeFrom(mci)
+	modifiedObj := mci.DeepCopy()
```

Comment:
We can only `deep copy` on needs. 
Something like:
if (no finalizer) {
    //deep copy and patch.
}